exp/internal/controllers/machinepool_controller_noderef.go

```diff
@@ -184,9 +186,11 @@ func (r *MachinePoolReconciler) getNodeReferences(ctx context.Context, c client.
 			continue
 		}
 		if node, ok := nodeRefsMap[providerID]; ok {
-			available++
 			if nodeIsReady(&node) {
 				ready++
+				if noderefutil.IsNodeAvailable(&node, *minReadySeconds, metav1.Now()) {
+					available++
+				}
```

Comment:
This changes the behavior, now we're defining an available node as a ready node first; is that on purpose?

Reply:
Ohh.... i think i have followed similar pattern as that of machinesets https://github.com/kubernetes-sigs/cluster-api/blob/07cd79e65f76a70e721329d7b99ea9fd8674e6a7/internal/controllers/machineset/machineset_controller.go#L889-L893

Reply:
It is a change in behavior but in my opinion it is a bugfix.

MachinePool API

```go
	// The number of available replicas (ready for at least minReadySeconds) for this MachinePool.
	// +optional
	AvailableReplicas int32 `json:"availableReplicas,omitempty"`
```

MachineDeployment API

```go
	// Total number of available machines (ready for at least minReadySeconds)
	// targeted by this deployment.
	// +optional
	AvailableReplicas int32 `json:"availableReplicas"`
```
(In MachineDeployments it is actually implemented as documented)

In my opinion the AvailableReplicas fields only make sense if they are not identical to the ReadyReplicas field.